src/Gui/BlenderNavigationStyle.cpp

```diff
@@ -252,18 +252,6 @@ SbBool BlenderNavigationStyle::processSoEvent(const SoEvent * const ev)
                 this->panningplane = vv.getPlane(viewer->getSoRenderManager()->getCamera()->focalDistance.getValue());
                 this->lockrecenter = false;
             }
-            else {
```

Comment:
When rotating an object in the 3d view then the pivot point is always the center of the view. So, we need a way for the user to set this pivot point. For me it would be also OK if the user e.g. has to press additionally the Shift key when doing a single middle click.

Reply:
Yes I'm a bit uncomfortable too with disabling middle-click entirely... Users will complain that this is not "Blender" anymore... The thing is, if you just click+release the middle mouse button, the view is centered. If you click, drag and release, you are performing the correct action. It's easy to make it wrong... Shift+middle-click is used for panning already in this mouse model if I'm not mistaken... Ctrl+middle-click maybe?

Reply:
I agree. Middle button click behavior should stay as is and the end user should learn to use it correctly (click+release sets rotation focus and click+drag+release is used for rotation). Introducing modifier key is in my opinion not desirable as a few other navigation styles use the same middle button click+release to set rotation focus. In addition the most common mouse operations should stay achievable without using the keyboard in Blender navigation style.

Reply:
If keep middle-click is a must, we can keep this functionality. However I'm looking the [Blender Navigation docs](https://docs.blender.org/manual/en/dev/editors/3dview/navigate/align.html) and I can't see any reference for middle-click to set the rotation pivot point.
Maybe View Selected(NumpadPeriod) or Center View to Cursor(Alt-Home) fit our needs.
Another solution could be replicate Maya gestures behavior; pan is only triggered if middle-click is done inside an object so it may help preventing accidental clicks.

Reply:
FreeCAD and Blender are two different software. Therefore better term i guess would be Blender alike. We in the end are not after 100% feature parity as that likely wouldn't be possible to achieve anyway. Due to differences between FreeCAD and Blender. If you will hover over the navigation indicator in the bottom right area of the status bar. Some clues will be presented in the tooltip. And as you can see in FreeCAD more navigation styles have the middle button click option for setting rotation focus. Therefore i guess this is more of an FreeCAD feature in the end. And what you describe as an issue (jumping displacement) is actually a feature (setting rotation the focus). Disabling this feature is a no go in my opinion and should be reverted before PR gets merged.

Reply:
That make sense. I will generate a PR keeping the middle-click feature.
I Hope it works for everyone.